### PR TITLE
Fix simplex.py

### PR DESCRIPTION
**Pull request by imengus**
### Describe your change:

I discovered that the algorithm I used initially gives suboptimal solutions when a pivot on a slack column is required; the algorithm terminates prematurely.

This pull request changes function inputs and doctests to reflect changes made to the steps of the algorithm. Mainly:
- Tableau __init__ takes arg n_art_vars
- special case added in run_simplex docstring

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by imengus** [comment]
Having found an error in the course that taught this simplex algorithm, I ask you to review my changes once more, @cclauss

File: linear_programming/simplex.py [hunk 1]
```diff
@@ -20,31 +20,28 @@
...
 
-    >>> t = Tableau(np.array([[-1,-1,0,0,-1],[1,3,1,0,4],[3,1,0,1,4.]]), 2)
+    >>> Tableau(np.array([[-1,-1,0,0,-1],[1,3,1,0,4],[3,1,0,1,4.]]), 2, 2)
     Traceback (most recent call last):
     ...
     ValueError: RHS must be > 0
     """
 
-    def __init__(self, tableau: np.ndarray, n_vars: int) -> None:
+    def __init__(self, tableau: np.ndarray, n_vars: int, n_art_vars: int) -> None:
```
Comment (cclauss) [inline linear_programming/simplex.py:29]: What does `art` mean in this context and why does the reader have to guess?  Please use real words even it if makes the variable longer.  https://word.tips/words-start-with/art

File: linear_programming/simplex.py [hunk 2]
```diff
@@ -62,48 +59,35 @@ def __init__(self, tableau: np.ndarray, n_vars: int) -> None:
...
-        ['x1', 'x2', 's1', 's2', 's3', 'a1', 'RHS']
-
-        >>> Tableau.generate_col_titles()
-        Traceback (most recent call last):
-        ...
-        ValueError: Must provide n_vars, n_slack, and n_art_vars
-        >>> Tableau.generate_col_titles(-2, 3, 1)
-        Traceback (most recent call last):
-        ...
-        ValueError: All arguments must be non-negative integers
```
Comment (cclauss) [inline linear_programming/simplex.py:79]: Do we NEED to remove these tests?!?

Comment (imengus) [reply linear_programming/simplex.py:79]: generate_col_titles was no longer a static method. I added another doctest for Tableau class, however

**Comment by imengus** [comment]
@CaedenPH sorry about the number of commits; I had issues with ruff etc. Could you please check this version?

**Comment by imengus** [comment]
To clarify, the old version couldn't optimally solve certain problems like the one I inserted on lines [270-286](https://github.com/imengus/Python/blob/master/linear_programming/simplex.py#L270-L286) in a doctest. I only realised this after that version was merged, given that it is a rare case. I adapted the implementation of other courses to fix this issue, but took the opportunity to add more doctests and checks.

**Review by CaedenPH — APPROVED** [review]
Nice change, thanks for adding the extra doctests.

@tianyizheng02 Pinging for review

**Review by tianyizheng02 — APPROVED** [review]
Honestly I'm not super familiar with the simplex algorithm, so I can't confirm whether this fixes the implementation—but the code itself looks good to me

**Pull request merged**